src/domain/service_domain_db.erl

```diff
@@ -0,0 +1,141 @@
+-module(service_domain_db).
+
+-behaviour(mongoose_service).
+
+-include("mongoose_config_spec.hrl").
+-include("mongoose_logger.hrl").
+
+-define(GROUP, service_domain_db_group).
+
+-export([start/1, stop/0, config_spec/0]).
+-export([start_link/0]).
+-export([enabled/0]).
+-export([force_check_for_updates/0]).
+-export([sync/0]).
+
+%% gen_server callbacks
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2,
+         terminate/2, code_change/3]).
+
+%% ---------------------------------------------------------------------------
+%% Client code
+
+start(Opts) ->
+    mongoose_domain_sql:start(Opts),
+    mongoose_domain_db_cleaner:start(Opts),
+    ChildSpec =
+        {?MODULE,
+         {?MODULE, start_link, []},
+         permanent, infinity, worker, [?MODULE]},
+    supervisor:start_child(ejabberd_sup, ChildSpec),
```

Comment:
it could be.

Reply:
Cleaner does nothing most of the period of time, so would never trigger intensity limit.

Fail in the initial init of the service should crash everything anyway.

The only other thing the service does is calling mongoose_domain_loader:check_for_updates, which has `try...catch all` in it, so not likely to fail.

Reply:
I understand, but I really think that if the supervision tree has a clean hierarchy, it is easier to reason about it. A lat tree makes it more difficult to figure out where a process comes from. But it can stay as it is now, I am just using tools like Observer from time to time and it is really nice when supervision tree reflects the structure of the application.